File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int prefixMin = nums[0], maxDiff = -1, d;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == prefixMin) {
                continue;
            }
            d = nums[i] - prefixMin;
            if (d > maxDiff) {
                maxDiff = d;
            }
            if (nums[i] < prefixMin) {
                prefixMin = nums[i];
            }
        }
        return maxDiff;
    }
}

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        prefix_min = nums[0]
        max_diff = -1
        for i in range(1, len(nums)):
            if nums[i] == prefix_min:
                continue
            d = nums[i] - prefix_min
            if max_diff < d:
                max_diff = d
            if nums[i] < prefix_min:
                prefix_min = nums[i]
        return max_diff